cmd/statusd/utils.go

```diff
@@ -1397,3 +1394,39 @@ func testValidateNodeConfig(t *testing.T, config string, fn func(common.APIDetai
 
 	fn(resp)
 }
+
+// ensureNodeSync waits until synchronzation is done and more safe
+// than only waiting a fixed amount of time.
+func ensureNodeSync(t *testing.T, nodeManager common.NodeManager) {
+	start := time.Now()
+	wait := 5 * time.Second
+	les, err := nodeManager.LightEthereumService()
+	if err != nil {
+		t.Error(err)
+	}
+	if les == nil {
+		t.Errorf("LightEthereumService is nil")
+	}
+
+	for {
+		// Some time needed even initially.
+		time.Sleep(wait)
+
+		downloader := les.Downloader()
+
+		if downloader != nil {
+			isSyncing := downloader.Synchronising()
+			progress := downloader.Progress()
+
+			if !isSyncing && progress.HighestBlock > 0 && progress.CurrentBlock >= progress.HighestBlock {
+				break
+			}
+		}
+
+		if time.Now().Sub(start) > (5 * time.Minute) {
```

Comment:
It would be better if a caller can control it via a context maybe.

Reply:
By passing a timeout context? Always more nice than constants. Good hint.